hack/dev-setup-vagrant

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+#
+# Copyright 2018 The Gardener Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DEV_DIR=$(dirname "${0}")/../dev
+EXAMPLE_DIR=$(dirname "${0}")/../example
+MINIKUBE_IP=$(minikube ip)
+IP_ROUTE=$(ip route get 1)
+IP_ADDRESS=$(echo ${IP_ROUTE#*src} | awk '{print $1}')
+MINIKUBE_SEED_KUBECONFIG=${DEV_DIR}/minikube-seed-kubeconfig
+
+kubectl apply -f ${EXAMPLE_DIR}/cloudprofile-vagrant.yaml
+kubectl apply -f ${EXAMPLE_DIR}/secret-dev-vagrant.yaml
+kubectl apply -f ${EXAMPLE_DIR}/privatesecretbinding-core-vagrant.yaml
+
+kubectl config set-cluster gardener-dev \
+  --kubeconfig ${MINIKUBE_SEED_KUBECONFIG} \
+  --certificate-authority $HOME/.minikube/ca.crt \
+  --server https://${MINIKUBE_IP}:8443 \
+  --embed-certs=true
+
+kubectl config set-credentials gardener-dev \
+  --kubeconfig ${MINIKUBE_SEED_KUBECONFIG} \
+  --client-certificate $HOME/.minikube/client.crt \
+  --client-key $HOME/.minikube/client.key \
+  --embed-certs=true
+
+kubectl config set-context gardener-dev \
+  --kubeconfig ${MINIKUBE_SEED_KUBECONFIG} \
+  --cluster gardener-dev \
+  --user gardener-dev
+
+kubectl config use-context gardener-dev \
+  --kubeconfig ${MINIKUBE_SEED_KUBECONFIG}
+
+# Different base64 implementations have different flags
+MINIKUBE_SEED_KUBECONFIG_B64=$(base64 ${MINIKUBE_SEED_KUBECONFIG} | tr -d '\r\n')
+
```

Comment:
Not sure about the below three `cat` commands. Can't we copy the files from the `example` folder and replacing the necessary fields (as far as I can see we only need to replace one value per resource) via `jq` or `sed` or other means?
We should avoid duplicating the specification and the need to double/triple maintain it.